organisations/models.py

```diff
@@ -112,8 +114,18 @@ class Meta:
 
     @transaction.atomic
     def delete(self, *args, **kwargs):
+        # Delete the related event contact info before deleting the user
         self.delete_related_p_event_contact_info()
-        return super().delete(*args, **kwargs)
+        # Make a log entry for the deletion
+        LogEntry.objects.log_create(
+            self, force_log=True, action=LogEntry.Action.DELETE
+        ).save()
+        # Delete the user without creating a auditlog entry,
+        # because the log entry is already created above and
+        # there would be a foreign key constraint violation
+        # if the log entry is not created before the user is deleted.
+        with disable_auditlog():
+            return super().delete(*args, **kwargs)
```

Comment:
Can there be any other related objects that are cascade deleted here? If there can be, then shouldn't their deletions still be audit logged here?

Reply:
That was my concern too, but the GDPR related tests should check these. The related data is "anonymized"  before calling the user delete. To be honest, I could not find any other way to do this.

Reply:
Do you mean `test_delete_profile_data_from_gdpr_api`? So is it so that the anonymization changes to related enrolments, study groups and event queue enrolments are not logged here now?

Reply:
This is something that would be good to look into / check what's going on here really. And the GDPR tests should be fixed as they're at least partially broken at the moment (i.e. the snapshot contains GDPR data when it should be anonymized). But both of these things can be done outside of this PR (or in this PR if you want to, but not necessary).